packages/api/src/providers/fitbit.ts

```diff
@@ -347,7 +354,7 @@ export class Fitbit extends Provider implements OAuth2 {
     const weight = resWeight.status === "fulfilled" ? resWeight.value : undefined;
 
     // User has body info, so we want to keep processing if only `weight` is missing
-    if (!user) {
+    if (containsProfile && !user) {
       if (!weight) throw new Error("All Requests failed");
       throw new Error("User Request failed");
     }
```

Comment:
Don't we need to rewrite this logic? If there was no `profile` scope and it failed to get the weight, shouldn't we throw the `All Requests failed` error?

The way it is, in a situation like that ^ we'd return a payload with only `metadata` - considering the change we did to `mapToBody` as well.

Reply:
For consistency with other errors i left it for only if they all fail

Reply:
Sounds good WRT to the error handling. 👍 

But what about `mapToBody` returning only metadata w/o actual data, do we leave as is or should we not include this provider in the response in that case?

Reply:
Yea i see the issue, adding a tweak so that it throws error if weight is false and containsProfile is false as well